devtools/maven/src/main/java/io/quarkus/maven/AbstractImageMojo.java

```diff
@@ -26,12 +27,22 @@ public enum Builder {
     @Parameter(defaultValue = "docker", property = "quarkus.container-image.builder")
     Builder builder = Builder.docker;
 
+    @Parameter(property = "quarkus.container-image.dry-run")
+    boolean dryRun;
+
     @Override
     protected boolean beforeExecute() throws MojoExecutionException {
         systemProperties.put("quarkus.container-image.builder", builder.name());
         return super.beforeExecute();
     }
 
+    @Override
+    public void execute() throws MojoExecutionException, MojoFailureException {
+        if (!dryRun) {
```

Comment:
If it is a dry run.. you should output the properties that could/would have been set. This is your chance to understand what would have happened...